Training/Training/intlist.h

```diff
@@ -0,0 +1,38 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment2 ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// inlist.c
+// Header file for creating a Linked List
+// ------------------------------------------------------------------------------------------------
+#pragma once
+#include <stdbool.h>
+
+#define E_INDEX_OUT_OF_RANGE  -1
+#define E_EMPTY_LIST -2
+#define E_LIST_DOES_NOT_EXIST -3
+#define E_VALUE_NOT_FOUND -4
+
+typedef int dataType; //the datatype can be changed based on requirement

```

Comment:
Good. But there are some other things you have to take care of, to make this truly generic.